mtags/src/main/scala-3/scala/meta/internal/pc/completions/MatchCaseCompletions.scala

```diff
@@ -159,20 +161,48 @@ object CaseKeywordCompletion:
           val ts = s.info.dealias.typeSymbol
           if (isValid(ts)) then visit(ts, ts.decodedName, Nil)
         )
-
       // Step 2: walk through known subclasses of sealed types.
-      MetalsSealedDesc.sealedStrictDescendants(selectorSym).foreach { sym =>
+      val sealedDescs = MetalsSealedDesc.sealedStrictDescendants(selectorSym)
+      sealedDescs.foreach { sym =>
         val autoImport = autoImportsGen.forSymbol(sym)
         autoImport match
           case Some(value) =>
             visit(sym, sym.decodedName, value)
           case scala.None =>
             visit(sym, sym.showFullName, Nil)
       }
+      val res = result.result()
+
+      selector match
+        // In `List(foo).map { cas@@} we want to provide also `case (exhaustive)` completion
+        // which works like exhaustive match.
+        case EmptyTree =>
```

Comment:
[no action required]
It seems a bit hacky to use`EmptyTree` for representing the case `List(foo).map { cas@@` (though it should be fine because there're no other cases where we can put `case` keywords).

It might be easier to understand if we receive something like `expr: Tree` instead of `selector: Tree` where `expr` can represent both `expr match { ... }` and `expr { case ... }`

but nevermind, I'm not sure it's eligible and not sure this really improves the readability of the code, no need to do something in this PR 😅

Reply:
[nit]
maybe, is it possible to change the signature of `contribute` method from `selector: Tree` into `selector: Select | EmptyTree`?

Reply:
I'm also not sure about these changes, I will try but maybe in different pr?